crates/net/network/src/manager.rs

```diff
@@ -865,6 +865,44 @@ where
         }
     }
 
+    /// Returns [`PeerInfo`] for all connected peers
+    fn get_peer_infos(&self) -> Vec<PeerInfo> {
+        let peer_manager = self.swarm.state().peers();
+        let mut peers = Vec::with_capacity(peer_manager.num_known_peers());
+        for (peer_id, session) in self.swarm.sessions().active_sessions() {
+            if let Some((record, kind)) = peer_manager.peer_by_id(*peer_id) {
+                peers.push(session.peer_info(&record, kind));
+            }
+        }
+        peers
+    }
+
+    /// Returns [`PeerInfo`] for a given peer.
+    ///
+    /// Returns `None` if there's no active session to the peer.
+    fn get_peer_info_by_id(&self, peer_id: PeerId) -> Option<PeerInfo> {
+        self.swarm.sessions().active_sessions().get(&peer_id).and_then(|session| {
+            self.swarm
+                .state()
+                .peers()
+                .peer_by_id(peer_id)
+                .map(|(record, kind)| session.peer_info(&record, kind))
+        })
+    }
+
+    /// Returns [`PeerInfo`] for a given peers.
+    ///
+    /// Ignore the non-active peer.
+    fn get_peer_infos_by_ids(&self, peer_ids: impl IntoIterator<Item = PeerId>) -> Vec<PeerInfo> {
+        let mut infos = Vec::new();
+        for peer_id in peer_ids {
+            if let Some(info) = self.get_peer_info_by_id(peer_id) {
+                infos.push(info)
+            }
+        }
+        infos
```

Comment:
this can be solved via an iterator `.and_then + collect`

Reply:
excuse me, did you mean the `.filter_map + collect`?

Reply:
yep